tests/bot/cogs/sync/test_cog.py

```diff
@@ -142,20 +148,32 @@ async def test_sync_cog_on_guild_role_create(self):
             "permissions": 8,
             "position": 23,
         }
-        role = helpers.MockRole(**role_data)
+        role = helpers.MockRole(**role_data, guild=self.guild_id)
```

Comment:
You have to pass `MockGuild` here and adding this ID after fixing cog L50's issue.